.github/workflows/rust.yml

```diff
@@ -102,13 +100,13 @@ jobs:
 
       - name: Check formatting
         working-directory: ${{ matrix.directory }}
-        run: cargo +${{ matrix.toolchain }} make format -- --check
+        run: just format
```

Comment:
I would have expected to still see a check flag passed through to this.

Is there a reason we've moved away from that? Formatting and checking formatting are different things in my mind

Reply:
How much prettier? (Just asking for your judgement, you don't have to supply a demo)

My instinct is that this isn't worth the change, there's a command designed to specifically check if things are formatted, and that's what we're doing here. Generally speaking, rewriting checked in source files in CI is not a good approach, it can cause problems with race conditions, and it's harder to debug. 

This isn't just a theoretical mind you, it's a genuine problem we're already seeing with our `codegen` steps in our blocks, where we need to format after codegen and because it's rewriting the source files in CI it's causing problems and we've had to `gitignore` them. 

I'm not saying that we _will_ run into issues here from it, I just don't see a reason to risk it. I also assume that actually formatting will be slower than checking? Although probably not significantly